StreamChatClient_v3/Controllers/Controller.swift

```diff
@@ -27,20 +53,16 @@ public extension ControllerRemoteActivityDelegate {
 }
 
 /// A helper protocol allowing calling delegate on a given dispatch queue.
-protocol DelegateCallbable {
+protocol DelegateCallable {
```

Comment:
The description says `calling delegate on a given dispatch queue` but **DelegateCallable** doesn't declare any queue. 

Maybe it makes sense to adjust the implementation it in the following way (or change the docs):
```swift
protocol DelegateCallable {
    associatedtype Delegate
    var anyDelegate: Delegate { get }
    var callbackQueue: DispatchQueue { get }
}

extension DelegateCallable {
    func delegateCallback(_ callback: @escaping (Delegate) -> Void) {
        let provideDelegate = { [anyDelegate] in
            callback(anyDelegate)
        }

        if callbackQueue == .main, Thread.current.isMainThread {
            provideDelegate()
        } else {
            callbackQueue.async(execute: provideDelegate)
        }
    }
}
```
or 
```swift
protocol DelegateCallable {
    associatedtype Delegate
    var anyDelegate: Delegate { get }
}

protocol CallbackQueueCallable {
    var callbackQueue: DispatchQueue { get }
}

extension DelegateCallable where Self: CallbackQueueCallable { ... }
```

Reply:
I see what you mean. The main functionality is defined in `extension DelegateCallable where Self: Controller`. The protocol is really just "has delegate?". I'll fix the documentation.

Reply:
fixed in 1efc3851293b36a4bcb5479e8a25777b004ed240